pkgs/by-name/pk/pkl/package.nix

```diff
@@ -0,0 +1,71 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, gradle
+, jdk17
+}:
+stdenv.mkDerivation rec {
+  pname = "pkl";
+  version = "0.26.2";
+
+  src = fetchFromGitHub {
+    owner = "apple";
+    repo = "pkl";
+    rev = version;
+    sha256 = "sha256-Q7B6DRKmgysba+VhvKiTE98UA52i6UUfsvk3Tl/2Rqg=";
+    # the build needs the commit id, replace it in postFetch and remove .git manually
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      export commit_id=$(git rev-parse --short HEAD)
+      cp ${./set_commit_id.patch} set_commit_id.patch
+      chmod +w set_commit_id.patch
+      substituteAllInPlace set_commit_id.patch
+      git apply set_commit_id.patch
+      rm set_commit_id.patch
+      find "$out" -name .git -print0 | xargs -0 rm -rf
+    '';
+  };
+
+  nativeBuildInputs = [ gradle ];
+
+  mitmCache = gradle.fetchDeps {
+    inherit pname;
+    data = ./deps.json;
+  };
+
+  __darwinAllowLocalNetworking = true;
+
+  gradleFlags = [
+    "-x" "spotlessCheck"
+    "-DreleaseBuild=true"
+    "-Dorg.gradle.java.home=${jdk17}"
```

Comment:
Heads up: Pkl 0.28 will require Java 21 to build.

Java 21 can be used to build Pkl 0.26.2 as well, so, you might want to just switch to it now.